deep_qa/data/data_generator.py

```diff
@@ -46,6 +46,18 @@ class DataGenerator:
         certain padding values or not.  It's handled correctly for ``num_sentence_words`` and
         ``num_word_characters`` in :class:`~deep_qa.training.TextTrainer`, but you need to be sure
         to implement it correctly in subclasses for this to work.
+    permute_probability: double, optional (default=.2)
```

Comment:
You're right that if your data is adversarial, this will perform poorly.  I'm not too concerned about it in the typical case, though.